pkg/server/apis/v1alpha1/types.go

```diff
@@ -248,3 +248,40 @@ type StorageUsage struct {
 type StorageCleanup struct {
 	Paths []string `json:"paths"`
 }
+
+// ExecutionContextSpec describes the execution context
+type ExecutionContextSpec struct {
+	Cluster     string `json:"cluster"`
+	Namespace   string `json:"namespace"`
+	Integration string `json:"integration"`
+	PVC         string `json:"pvc"`
+}
+
+// ExecutionContextStatus describe the status of execution context, it contains information that affects
+// pipeline execution, like reserved resources, pvc status.
+type ExecutionContextStatus struct {
+	// Phase of the execution context, could be 'Ready', 'NotReady' or 'Unknown'
+	Phase             ExecutionContextPhase                `json:"phase"`
+	ReservedResources map[core_v1.ResourceName]string      `json:"reservedResources"`
+	PVC               *core_v1.PersistentVolumeClaimStatus `json:"pvc"`
+}
+
+// ExecutionContext represtents a context used to execute workflows.
+type ExecutionContext struct {
+	// Metadata for the particular object, including name, namespace, labels, etc
+	meta_v1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
I think ExecutionContext doesn't have concept like `namespace`, and also no need to use `labels`, it's not k8s resources, and we should not treat like that.

Reply:
OK，I will delete it.